UI changes and optimizations

rsiebert opened the pull request:
This PR improves the UI and contains internal optimizations
- The UI has been improved and ported to the halo styles
- The main view contains tabs for fast access to channels and recordings
- The actionbar is used throughout all screens
- More programs will automatically be loaded when the user reaches the end of program list
- The recordings are grouped into tabs
- Orientation changes remember the selected tab and items
- and much more...

Note: 
- This PR contains the PR #19, #20, #21 and #22.
- The miminum android SDK version is now 11.

Commit:
Remove the status icon.
The state of the recording can be seen by the
action bar tab name.

Commit:
Fixed the shown recording count in the actionbar subtitle after an
orientation change.

Commit:
Merge branch 'recording_layout' into development

Commit:
Show the channel icon instead of the application icon in the
actionbar of the program list.

Commit:
Show the channel icon instead of the application icon in the
actionbar of the program details screen.

Commit:
Added the action bar tabs to the channel list so the user can now
quickly navigate to the recordings or the upcoming program guide
screens.

Commit:
Removed the recordings menu item. 
The user can just use the actionbar tab.

Commit:
Removed an unused layout file.

Commit:
The selected theme is now applied again.

Commit:
Added transparency to the progressbar background.

Commit:
Move the handling of the settings menu into the main activity. 
This is common to all fragments in the tabs.

Commit:
Merge branch 'channel_layout' into development

Commit:
The theme is now set according to the user defined setting.

Commit:
Set the theme according to the user preference.

Commit:
Fixed setting the correct tab after coming back from another activity.
Made bundle identifier names unique.

Commit:
Improved the layout of the program details screen.

Commit:
Improved the layout of the recording details screen.

Commit:
Removed menu items and left only those that are common to all
activities.

Commit:
Added a separate channel menu.

Commit:
Added a separate recording menu.

Commit:
Moved the progressbar below the text and increased its margin to the
next list item.

Commit:
Moved the connect method from the channel fragment into the channel
activity.
The main menu contains the refresh menu item so handle the refresh in
here.

Commit:
The user can now refresh the list of available recordings from any
recordings screen.

Commit:
Removed the last tab which was called autorec.
All recordings from this tab will now be shown in the upcoming tab.

Commit:
Removed a warning when setting the icon for the actionbar.

Commit:
Removed a deprecation warning when setting the icon drawable.

Commit:
Fixed the channel icon size to 54dp.

Commit:
Don't recreate the channel list activity when the user has pressed the
home button.

Commit:
Moved the date processing into a separate helper class.

Commit:
Moved the series information processing into the helper class.

Commit:
Removed unused menu and layout files.

Commit:
Renamed the program menu file.

Commit:
Changed all references of programme to program.
This is just a type/spelling correction.

Commit:
Moved the connection method into the helper class.

Commit:
Moved the ChannelViewWrapper class into the new adapter package.

Commit:
Moved the recording adapter and view wrapper classes into separate
files.

Commit:
Moved the program adapter and view wrapper classes into separate files.

Commit:
Moved the channel adapter class into a separate file.

Commit:
Removed all warnings in the layout files by replacing hard coded strings
with translatable ones.

Commit:
Increased the padding to the right screen border. Also added the
duration to the layout.

Commit:
Increased the padding to the right screen border.

Commit:
Added the action bar to the search result activity and improved the
layout a little.

Commit:
Use the methods provided by the utils class.

Commit:
Small layout adjustments. Changed the color of the title string.

Commit:
Moved the view wrapper and adapter classes from the search result
activity file into separate files.

Commit:
Create a title font style.

Commit:
Removed references to the old custom theme.

Commit:
Removed an unused import.

Commit:
Removed unused images.

Commit:
The search icon is now visible in the action bar. 
Made the menu names more consistent (they now start with menu_).

Commit:
Removed more unused files.

Commit:
Changed the title of the settings activity.

Commit:
Made the search icon theme dependent.

Commit:
Updated the references to the search icon.

Commit:
Load the custom theme instead of the default one to include all
modifications.

Commit:
Made the title font theme dependent.

Commit:
Made the style of the headers above the actual text theme dependent.

Commit:
Show the menu and add the search logic in the recording fragment.

Commit:
Update the program list in case the activity was resumed.

Commit:
Added a program details menu to the program activity.
The intents are only created when the menu item was selected. The menu
items are updated when the new status has been reported by the
interface.

Commit:
Hide the play menu item.
Its not used in this activity.

Commit:
Added the details menu to the recording details activity.
This menu is also used by the program details screen. These changes are
the first steps towards code simplification and consolidation.

Commit:
Load new programs only when no other loading is in progress.

Commit:
Reduced the padding at the top.

Commit:
Fixed typo and increased the title font.

Commit:
Two consolidation changes.
- Ported the list activity to an activity so it and the program details
class can later derive from a common base class. 
- The context menu is read from a XML file. The logic is similar to the
one in the program details class.

Commit:
Moved common calls like record, cancel or delete programs and recordings
into separate methods.

Commit:
Use the common methods from the Utils class.

Commit:
Refactored the menu. Also use the Utils class methods.

Commit:
Moved the menu creation that is common to the programs and recordings
into the Utils class.

Commit:
Removed some Locale warnings.

Commit:
Renamed some variables only.

Commit:
Moved the retrieval of the selected theme id into the Utils class.

Commit:
Renamed the programs activity to program details activity to better
reflect its purpose.

Commit:
Renamed the recording activity to recording details activity to better
reflect its purpose.

Commit:
Converted the search list activity to an activity.

Commit:
Renamed the channel list layout file.
Its used by more than this class.

Commit:
Refactored the context menu by using the common classes from the Utils
class.

Commit:
Formatting changes only

Commit:
Refactored the channel list context menu to use a XML file.

Commit:
Renamed the details menu to context menu to better reflect its purpose.

Commit:
Fixed typos.

Commit:
Remember the selected channel tag after an orientation change.
The fragment management had to be changed so ensure that.

Commit:
Added a file which lists the items that still needs to be done.

Commit:
Do not show the channel icons in the actionbar if not required.

Commit:
Fixed some padding in the layouts when the channel icons are not
visible.

Commit:
Fixed the selection of the correct item in the context menu.
This was done by not storing the selected recording globally.
Additionally only the context menu selection method of the visible
fragment is called.

Commit:
Do not store the channel globally when the context menu was opened.

Commit:
Do not store the program globally when the context menu was opened.

Commit:
Removed unused import.

Commit:
Hide the play menu item when a program is scheduled. 
Its not yet recording and therefore can't be played.

Commit:
Allow playing the first item in the program list. 
This program is currently transmitted and can be played.

Commit:
Changed the state icon size and padding.

Commit:
Added the program state (recording, scheduled) to the details screen.

Commit:
Refactored the settings activity.
Used a listener to check if the connection settings have changed. The
current values (except the password) are also displayed.

Commit:
Increased the left and top margin of the channel and recording icons.

Commit:
Fixed sorting of the recordings.

Commit:
Moved some calculations into constant variables.

Commit:
There is no autorec actionbar tab. Remove the logic for it.

Commit:
Updated some strings and added its German translation.

Commit:
Fixed some typos.

Commit:
Removed some completed items from the todo file.

Commit:
Changed type to content type.

Commit:
Changed the search layout to be according to the other ones.
Added the separate content type and series information.

Commit:
Renamed the program list adapter class.

Commit:
Moved the handling of the state icon into a separate method in the Utils
class.

Commit:
Moved the handling of the duration into a separate method in the Utils
class.

Commit:
Moved the handling of the time into a separate method in the Utils
class.

Commit:
Moved the handling of the date into a separate method in the Utils
class.

Commit:
Moved the handling of the series information into a separate method in
the Utils
class.

Commit:
Moved the handling of the content types into a separate method in the
Utils class.

Commit:
Removed duplicate invalidate calls.

Commit:
Use the methods from the Utils class.

Commit:
Moved the handling of the channel icon and name into a separate method
in the Utils class.

Commit:
Made some objects and variables final and added blank method comments.

Commit:
Moved the handling of the description into a separate method in the
Utils class.

Commit:
Some small restructuring and used the setDescription method from the
utils class.

Commit:
Updated some methods to show or hide an additional textview.
This is primarily used in the details activities where each text has a
header.

Commit:
Changed the details activities to use the methods from the Utils class.

Commit:
Added the recording indication as a separate icon in the channel list
layout.

Commit:
Guard against possible null pointer exception.

Commit:
Consolidated the names of the views in the layouts.

Commit:
More guards against possible null pointer exceptions.

Commit:
Made the description view name consistent throughout all layouts.

Commit:
Fixed the blank method comments.

Commit:
Refactored the channel list adapter.
The wrapper class has been removed. The creation of the view and showing
the values have been moved into the adapter class.

Commit:
Refactored the search result adapter.
The wrapper class has been removed. The creation of the view and showing
the values have been moved into the adapter class.

Commit:
Refactored the recording list adapter.
The wrapper class has been removed. The creation of the view and showing
the values have been moved into the adapter class.

Commit:
Refactored the program list adapter.
The wrapper class has been removed. The creation of the view and showing
the values have been moved into the adapter class.

Commit:
Commented the old channel wrapper class until this activity has been
rewritten.

Commit:
Removed the invalidate calls. The adapter classes will handle the view
updates.

Commit:
Small layout changes.
Moved the content type to the top and added the series info above the
description.

Commit:
Updated the todo item list

Commit:
Honor the confirmation dialog when deleting a recording.

Commit:
Use padding to separate the list items even when the description is
hidden.

Commit:
Removed completed items from the todo list.

Commit:
Added first implementation of the status fragment.
Currently only the total and free disc space is shown.

Commit:
Convert the disc space usage to megabytes.

Commit:
Add some extra strings to the disk space status.

Commit:
Reduce the number of linear layouts and make the main layout responsible
for the bottom padding.

Commit:
Disabled the the program guide tab for now.

Commit:
Updated the readme file and added some screenshots.

Commit:
Update README

Commit:
Fixed the links to the screenshots in the readme file.

Commit:
Update README

Commit:
Updated the readme file and removed the screenshots.
The files are located in a separate branch.

Commit:
Merge branch 'development' of git://github.com/rsiebert/TVHGuide into development

Conflicts:
	README

Commit:
Moved the disc space logic into a separate method.

Commit:
Added the available number of channels and recordings to the status
page.

Commit:
Show the number of search results in the action bar.

Commit:
Update the icon if the program status changes.

Commit:
Added the status icon to the recording details screen.

Commit:
Updated the todo list.

kecksk0enig commented on src/org/tvheadend/tvhguide/ChannelListTabsActivity.java:
```diff
@@ -0,0 +1,177 @@
+                // Detach the channel list fragment, because another will be attached
+                Fragment prevFragment = getFragmentManager().findFragmentByTag(tab.getText().toString());
+                if (prevFragment != null) {
+                    ft.detach(prevFragment);
```
detach ist still API-Level 13!
http://developer.android.com/reference/android/app/FragmentTransaction.html#detach%28android.app.Fragment%29

rsiebert replied:
Thx for the information. I have changed the min SDK version to 13.

Commit:
Set the android SDK to 17

Commit:
The minimum android SDK version is now 13.

Commit:
The minimum android SDK version is now 13.

Commit:
Bump the application version.

Commit:
Updated the todo file

Commit:
Added the information what is currently being recorded.

Commit:
Updated the todo file.

Commit:
Set the eclipse project android SDK version to 17.

Commit:
Hide the currently recording view if nothing is being recorded.

Commit:
Merge branch 'status' into development

Commit:
Consolidated existing and removed unused strings.

Commit:
Removed the class comment. The author is already mentioned in the
header.

Commit:
Removed the class comments and added me to the header. 
The original author is still mentioned in the header.

Commit:
Show a note to the user to restart the app when he switches the theme.

Commit:
Removed deprecation warning.

Commit:
Updated the todo file.

Commit:
Removed some warnings.

Commit:
Use the list_layout for the recordings.

Commit:
Enable fast scroll for all lists.
The user can quickly navigate to the desired position in the list.

Commit:
Increased the padding for the texts in the status page.

stv0g commented:
Wow, that looks fantastic.

Do you want to provide us with compiled APK in the gh-pages branch for example?
I've no Android SDK setup on my systems..

Thank you :-)

john-tornblom commented:
Yes, very nice indeed! Unfortunately, I'm extremely busy these days so it might take a while until I can have a look at the code. In addition, I'm still using Android 2.3 so I can't really test it thoroughly. Is it at all possible to keep the code backwards compatible? If not, I guess it gives me a reason to by some new hardware :)

ghost commented:
Welcome
Can you add Polish language to the application and issue a version of the android market? The other issue I added Polish translation. Thank you.
Sorry for off topic :)

Commit:
Fixed some string translations.

Commit:
Increased version number.

kecksk0enig commented:
@john-tornblom
I think there is no way to keep it backwardcompatible without loosing lot of new gui features. Maybe there are some emulation layer one coulde use to get an almost equal looking and feeling feature but it would really bloat the code.
Are thhere any statistics how many useres of tvhguide are still on androd 2.x?
I currently do run a custom buildd of tvhguide iincluding those changes from rsiebert and my own channellogo update and it works really well.

@piterEL
If you are a nativespeaker of polish feel free to get a copy of values.xml and provide an polish translation by your self. You won´t even need a fully ide, just a texteditor would work.

ghost commented:
I translated xml i this issue
https://github.com/john-tornblom/TVHGuide/pull/24

Commit:
Added polish translation.

Commit:
Updated the todo file.

Commit:
Updated the todo file.

Commit:
Merge branch 'development' into release

Conflicts:
	src/org/tvheadend/tvhguide/ChannelListTabsActivity.java

rsiebert commented:
@PiterEL I have added the polish translation to the development and release branch. But there are many more strings that need translation.

@john-tornblom It could be possible to use the sherlock actionbar to support android 2.2 but I don't know how much effort this would be. It would be nice to see how many people still use this app with android 2.2.

@all I have added an unsigned apk (version 1.7.1) to the gh-pages branch.

ghost commented:
Which strings ? Sorry i am beginner :) Tell me what isn't translated?

Commit:
Added an activity that selects the correct player class and the
specified playback preferences.

Commit:
Make use of the new player selection class.

Commit:
Use the values that was given by the player selection class and not from
the settings. Removed the call to the external player class.

Commit:
Use the values that was given by the player selection class and not from
the settings. Added a confirmation dialog before going to the google
play store.

Commit:
Added new strings.

Commit:
Updated the file to show separate preference screens for programs and
recordings.

ghost commented:
When can we download from google play version 1.7.1 ? Tomorrow i transleted to polish other string.xml

Commit:
Pass the name of the channel or the recording title to the
internal playback activity.

Commit:
Show the title that was given to us. Do not check for a channel.

Commit:
Added the play menu logic. It was missing in the context menu selection.

Commit:
Fixes wrong dependency.

Commit:
Added the playback selection activity to the manifest.

Commit:
In case no external media player was found a confirmation dialog is now
shown.

john-tornblom commented:
@rsiebert  About 8% of all active installs are on Android 2.x. Google have [a tutorial on how to keep backwards comparability](http://developer.android.com/training/backward-compatible-ui). I'm not sure if it's feasible, what do you think?

Commit:
Added the settings to the menu.

Commit:
Merge branch 'playback_settings' into development

Commit:
Updated polish translation.

Commit:
Added the helper class that manages the database handling.

Commit:
Initialize the database helper and remove the hardware key logic.

Commit:
Add the required icons and their styles.

Commit:
Add required strings.
They still need some tuning.

Commit:
Add the model which contains all relevant connection information.

Commit:
Move the fragment into the settings activity class.

Commit:
Add the menu that contains the add connection item for the action bar.

Commit:
Add the menu that contains the save and cancel items for the action bar.

Commit:
Add the context menu for a connection item.

Commit:
Added the layout for a connection. 
Still needs more love.

Commit:
Removed the connection options from the main screen and moved it into a
separate preference screen.

Commit:
Added the manage and the adapter that will show all available
connections in a list.

Commit:
Added the class that allows adding a new or editing an existing
connection.
This is still incomplete.

Commit:
Ported the connect method to use the database.

Commit:
Added the two activities to the manifest file.

Commit:
There is still a lot of work to be done.

Commit:
Allow only numbers to be entered for the port.

Commit:
Refactored the add connection class.
Still not done.

Commit:
Fix segfault when no selected connection is available.

Commit:
Remove the default values except the port number.

Commit:
Fix the switching of the selected connection.

Commit:
More refactoring of the class to add or edit a connection.
Pretty much complete now.

Commit:
The id is of the type long.

Commit:
More fixes. Still not done.

Commit:
More refactoring. 
Use a connection object to set and get the values.

Commit:
Changed some strings.

Commit:
Removed the texts and changes the selected indication to an image.

Commit:
The auto formatter did this.

Commit:
Added some validation to the add connection activity.

Commit:
Show or hide the selected image.

Commit:
Finalize the add or edit connection activity.

Commit:
Fixed the default port and removed the logging of the connection
details.

Commit:
Fixed the default port.

Commit:
Add a deselect menu item.

Commit:
Add the logic to deselect a connection.

Commit:
Show the light or dark version of some icons depending on the theme.

Commit:
Changed some strings and ids.

Commit:
Changed some ids and strings.

Commit:
Hide the refresh menu item if no connection is set active.

Commit:
Removed some logging statements. Show either the activate or deactivate
menu item.

Commit:
Hide the refresh menu item if no connection is set active.

Commit:
Improved the logic to show or hide the refresh menu item.

Commit:
Clear the list of channels and recordings when no connection is
available.

Commit:
Removed some logging.

Commit:
The active connection can now be chosen with a regular click.
Moved some logic into a separate method.

Commit:
Send back the result to reconnect when the activity is closed.

Commit:
Force a reconnect if the user returns form the connection settings
activity.

Commit:
Remove the unused onPause method.

Commit:
Reload the recordings by forcing a reconnect when the user returns from
the connection settings activity.

Commit:
Added new and theme dependent icons for the menus.

Commit:
Version bump.

Commit:
Merge branch 'multiple_servers' into development

Commit:
Updated the polish translation.

rsiebert commented:
@john-tornblom I think I'll give it a shot. I was busy implementing the multiple account functionality. Porting to android 2.2 is now on the list.

Commit:
Merge branch 'development' into release

rsiebert commented:
Maybe I should start squashing these commits. :-)

john-tornblom commented:
@rsiebert great! I got a few minutes over last weekend and had a quick look. Apparently, it is possible to use external libs (provided by google), see http://developer.android.com/tools/support-library/setup.html for the details.

Commit:
Merged branch development.

rsiebert commented:
@john-tornblom New TVHGuide version 1.8 is out. I have ported the application to be compatible with android 2.2. It uses the android support library and the preference fragment from https://github.com/kolavar/android-support-v4-preferencefragment. Works great on the HTC mini with Android 2.2 and on mine with 4.3.

ghost commented:
I installed your version(today) and i have a problem. I can't watch TV. When i click on play i have a ERROR OPENING FILE 66and other numer for example 5 , 77. On oficial app from GOOGLE PLAY i can watch TV with no problem... Please HELP

rsiebert commented:
I'll check this. You can change in the settings that an external media player shall be used.

ghost commented:
I use an external player name BSplayer and it works great on oficial TVHguide from google play.

Commit:
Merged branch development
